spec2sdk/generators/identifiers.py

```diff
@@ -1,21 +1,19 @@
 import re
 from keyword import iskeyword
+from typing import Final, Pattern
 
 from humps import decamelize, pascalize
 
+INVALID_CHARACTERS_PATTERN: Final[Pattern] = re.compile(r"[^0-9a-z]+|^[^a-z]+", flags=re.IGNORECASE)
```

Comment:
What about `_` and `-`? Will the new regex lead to the same names as the old one?
What about collisions like `abc-0` and `abc_0`? Will they lead to conflicting identifiers? Does this need to be handled somehow?

Reply:
`abc-0` and `abc_0` will produce the same result `abc_0`. This module only generates valid Python identifiers. Conflict checking is not part of it's responsibility. It can be handled in the code generators but it's not for now. I'm gradually add tests for the different parts of the library and rewrite corresponding code. When I get to the appropriate code, I'll add name conflict checks.

Reply:
> Will the new regex lead to the same names as the old one?

Yes, in almost all cases. The only exception is when variable ends with invalid characters. Old generator replaced trailing invalid characters with underscore, new one removes trailing characters.

Since nobody uses this library for now except us it's safe to change this part of the code.

Reply:
> When I get to the appropriate code, I'll add name conflict checks.

Maybe pydantic will also detect naming conflicts automatically and one does not even have to add any extra checks?

Reply:
> Maybe pydantic will also detect naming conflicts automatically and one does not even have to add any extra checks?

Can't tell right now. It also depends on the scope. For example, some identifiers are only used as method parameters, and it doesn't make sense to make them unique across all identifiers.